In [ ]:
import pandas as pd
import numpy as np

In [ ]:
PATH = '/content/datos_temperaturas.xlsx'
df = pd.read_excel(PATH, skiprows=1)

df.head(5)

X  Noise       Y
0   5.36   1.00  43.448
1  18.55  -0.36  64.742
2  13.21   0.65  56.948
3  11.68   0.86  54.572
4   9.25  -0.44  47.858

<p>Con el método .info() podemos observar que el dataset está formado por 3 columnas y 136 filas, cuyos valores son de tipo float. Además, es importante anotar que, el dataset no tiene valores faltantes, por tanto, no hay necesidad de aplicar tratamiento de <i>missing</i>.</p>

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 136 entries, 0 to 135
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   X       136 non-null    float64
 1   Noise   136 non-null    float64
 2   Y       136 non-null    float64
dtypes: float64(3)
memory usage: 3.3 KB


In [ ]:
df.columns = ['Dato 1', 'Dato 2', 'Dato 3']

<p>Con el método .head() podemos tener un primer vistazo de lo que contiene el dataset, de esta manera, conocemos de forma ordenada la información con la que vamos a tratar, así mismo, lograr sacar algunas conclusiones previas.</p>

In [ ]:
df.head(5)

Dato 1  Dato 2  Dato 3
0    5.36    1.00  43.448
1   18.55   -0.36  64.742
2   13.21    0.65  56.948
3   11.68    0.86  54.572
4    9.25   -0.44  47.858

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 136 entries, 0 to 135
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Dato 1  136 non-null    float64
 1   Dato 2  136 non-null    float64
 2   Dato 3  136 non-null    float64
dtypes: float64(3)
memory usage: 3.3 KB


In [ ]:
df['Dato 3'].mean()

np.float64(54.96945588235294)

<h2>Función para calcular la media aritmética usando el ciclo for</h2>
<p>Definimos una función calcular_media para que recorriendo todos los valores dentro de la columna, pueda sumarlos uno tras otro, e ir realizando el conteo del total de numeros encontrados, de esta manera, con una simple operación, se divide la suma entre el contador para obtener el valor de la media aritmética.</p>

In [ ]:
def calcular_media(df, columna):
  suma = 0
  contador = 0
  for valor in df[columna]:
    suma += valor
    contador += 1
  return suma / contador if contador > 0 else None

<p>Se aplica la función a cada columna y se obtienen los valores correspondientes de la media aritmética, dando como resultado:</p>
<li>Para la columna "Dato 1": 12.67</li>
<li>Para la columna "Dato 2": 0.08</li>
<li>Para la columna "Dato 3": 54.97</li>

In [ ]:
media1 = calcular_media(df, 'Dato 1')
media2 = calcular_media(df, 'Dato 2')
media3 = calcular_media(df, 'Dato 3')

print(f"La media de la columna Dato 1 es: {media1}")
print(f"La media de la columna Dato 2 es: {media2}")
print(f"La media de la columna Dato 3 es: {media3}")

La media de la columna Dato 1 es: 12.675514705882355
La media de la columna Dato 2 es: 0.08529411764705885
La media de la columna Dato 3 es: 54.969455882352946


<p>La media de la columna "Dato 2" es muy inferior a las otras dos columnas, dado que sus valores oscilan entre -1 y 1. Por otro lado, la media de la columna "Dato 3" resultó ser más alta que las demás, puesto que, los valores de ésta son significativamente más grandes.</p>

In [ ]:
df_ins = pd.DataFrame(np.array([[media1], [media2], [media3]]))
x=df_ins.T
x.columns = ['Dato 1', 'Dato 2', 'Dato 3']
df = pd.concat([df, x], axis=0)

df.head(5)

Dato 1  Dato 2  Dato 3
0    5.36    1.00  43.448
1   18.55   -0.36  64.742
2   13.21    0.65  56.948
3   11.68    0.86  54.572
4    9.25   -0.44  47.858

In [ ]:
def calcular_media_filas(fila):
    suma = 0
    contador = 0
    for valor in fila:
        suma += valor
        contador += 1
    return suma / contador if contador > 0 else None

df['Media'] = df.apply(calcular_media_filas, axis=1)

df.head(5)

Dato 1  Dato 2  Dato 3      Media
0    5.36    1.00  43.448  16.602667
1   18.55   -0.36  64.742  27.644000
2   13.21    0.65  56.948  23.602667
3   11.68    0.86  54.572  22.370667
4    9.25   -0.44  47.858  18.889333

<h2>Función varianza utilizando ciclos</h2>

In [ ]:
def varianza_filas(fila):
    valores = fila.iloc[:3]
    n = len(valores)

    # Calcular media
    total = 0
    for x in valores:
        total += x
    media = total / n

    # Calcular suma de cuadrados
    suma_cuadrados = 0
    for x in valores:
        suma_cuadrados += (x - media) ** 2

    return suma_cuadrados / (n - 1)

df['varianza'] = df.apply(varianza_filas, axis=1)

<p>Guardar el dataframe dentro de un archivo excel</p>

In [ ]:
df.to_excel('datos_temperaturas_resuelto.xlsx')